README.md

```diff
@@ -40,199 +38,107 @@ If you need more information about setting up your development environment Subst
 [Getting Started](https://substrate.dev/docs/en/knowledgebase/getting-started/) page is a good
 resource.
 
+## High-Level Architecture
+
+This repo has support for bridging foreign chains together using a combination of Substrate pallets
+and external processes called relayers. A bridge chain is one that is able to follow the consensus
+of a foreign chain independently. For example, consider the case below where we want to bridge two
+Substrate based chains.
+
++---------------+                 +---------------+
+|               |                 |               |
+|     Rialto    |                 |    Millau     |
+|               |                 |               |
++-------+-------+                 +-------+-------+
+        ^                                 ^
+        |       +---------------+         |
+        |       |               |         |
+        +-----> | Bridge Relay  | <-------+
+                |               |
+                +---------------+
+
+The Millau chain must be able to accept Rialto headers and verify their integrity. It does this by
+using a runtime module designed to track Gradpa finality. Since two blockchains can't interact
```

Comment:
~~Gradpa~~ Grandpa
Also, is it `Grandpa` or `GRANDPA`? We seem to use both.

Reply:
I think we should be using `GRANDPA` everywhere.